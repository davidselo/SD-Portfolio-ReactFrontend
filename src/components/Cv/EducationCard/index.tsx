import React from 'react';
import {Box, Container, Typography} from '@mui/material';
import SchoolIcon from '@mui/icons-material/School';
import {CvSectionTitle} from 'components/Cv/CvSectionTitle';

interface Props {
    education: Array<Study>;
}

interface Study {
    institution: string;
    url: string;
    area: string;
    studyType: string;
    startDate: string;
    endDate: string;
    score: string;
    courses: string[];
}

export const EducationCard: React.FC<Props> = ({education}: Props) => {
    return (
        <>
            <CvSectionTitle title="education" iconComponent={<SchoolIcon />} />
            {education.map(item => {
                return (
                    <Container disableGutters key={item.studyType}>
                        <Box
                            className="cv-experience--places"
                            sx={{marginTop: '10px'}}
                        >
                            <Typography variant="body1">
                                {item.studyType}
                            </Typography>
                        </Box>
                        <Box className="cv-experience--places">
                            <Typography variant="body2">
                                {item.institution}
                            </Typography>
                        </Box>
                        <Box className="cv-experience--places">
                            <Typography variant="body2">
                                {item.startDate}-{item.endDate}
                            </Typography>
                        </Box>
                    </Container>
                );
            })}
        </>
    );
};

export default EducationCard;
