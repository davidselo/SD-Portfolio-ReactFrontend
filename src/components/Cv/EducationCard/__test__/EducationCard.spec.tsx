import React from 'react';
import EducationCard from 'components/Cv/EducationCard';
import '@testing-library/jest-dom';
import {render, waitFor, getByText} from '@testing-library/react';

describe('EducationCard Component', () => {
    test('Should match the Screenshot', () => {
        // Arrange
        // Act
        const careerMockData = {
            institution: 'Institution 1',
            url: 'Url',
            area: 'Area',
            studyType: 'Study type example',
            startDate: '01/01/2002',
            endDate: '01/01/2005',
            score: 'Example score',
            courses: ['Course 1', 'Course 2'],
        };
        const {container} = render(
            <EducationCard education={[careerMockData]} />,
        );

        // Assert
        expect(container).toMatchSnapshot();
    });
    test('Should match the Screenshot for two education items', () => {
        // Arrange
        // Act
        const careerMockData1 = {
            institution: 'Institution 1',
            url: 'Url',
            area: 'Area 1',
            studyType: 'Study type example 1',
            startDate: '01/01/2002',
            endDate: '01/01/2005',
            score: 'Example score 1',
            courses: ['Course 1', 'Course 2'],
        };
        const careerMockData2 = {
            institution: 'Institution 2',
            url: 'Url 2',
            area: 'Area 2',
            studyType: 'Study type example 2',
            startDate: '01/01/2002',
            endDate: '01/01/2005',
            score: 'Example score 2',
            courses: ['Course 3', 'Course 4'],
        };

        const {container} = render(
            <EducationCard education={[careerMockData1, careerMockData2]} />,
        );

        // Assert
        expect(container).toMatchSnapshot();
    });
    test('Render outcome produce the correct HTML', async () => {
        // Arrange
        // Act
        const careerMockData = {
            institution: 'Institution 1',
            url: 'Url',
            area: 'Area',
            studyType: 'Study type example',
            startDate: '01/01/2002',
            endDate: '01/01/2005',
            score: 'Example score',
            courses: ['Course 1', 'Course 2'],
        };
        const {container} = render(
            <EducationCard education={[careerMockData]} />,
        );
        // Assert
        await waitFor(() => {
            expect(getByText(container, 'EDUCATION')).toBeInTheDocument();
            expect(getByText(container, 'Institution 1')).toBeInTheDocument();
            expect(
                getByText(container, 'Study type example'),
            ).toBeInTheDocument();
        });
    });
});
