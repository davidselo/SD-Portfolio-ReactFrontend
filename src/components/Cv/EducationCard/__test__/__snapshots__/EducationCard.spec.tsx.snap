// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EducationCard Component Should match the Screenshot 1`] = `
<div>
  <h5
    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
  >
    <div
      class="MuiBox-root css-1xhj18k"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="SchoolIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M5 13.18v4L12 21l7-3.82v-4L12 17l-7-3.82zM12 3 1 9l11 6 9-4.91V17h2V9L12 3z"
        />
      </svg>
      <div
        class="MuiBox-root css-1y6ic72"
      >
        EDUCATION
      </div>
    </div>
  </h5>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="cv-experience--places MuiBox-root css-1xaekgw"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Study type example
      </p>
    </div>
    <div
      class="cv-experience--places MuiBox-root css-0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Institution 1
      </p>
    </div>
    <div
      class="cv-experience--places MuiBox-root css-0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        01/01/2002
        -
        01/01/2005
      </p>
    </div>
  </div>
</div>
`;

exports[`EducationCard Component Should match the Screenshot for two education items 1`] = `
<div>
  <h5
    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
  >
    <div
      class="MuiBox-root css-1xhj18k"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="SchoolIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M5 13.18v4L12 21l7-3.82v-4L12 17l-7-3.82zM12 3 1 9l11 6 9-4.91V17h2V9L12 3z"
        />
      </svg>
      <div
        class="MuiBox-root css-1y6ic72"
      >
        EDUCATION
      </div>
    </div>
  </h5>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="cv-experience--places MuiBox-root css-1xaekgw"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Study type example 1
      </p>
    </div>
    <div
      class="cv-experience--places MuiBox-root css-0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Institution 1
      </p>
    </div>
    <div
      class="cv-experience--places MuiBox-root css-0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        01/01/2002
        -
        01/01/2005
      </p>
    </div>
  </div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="cv-experience--places MuiBox-root css-1xaekgw"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Study type example 2
      </p>
    </div>
    <div
      class="cv-experience--places MuiBox-root css-0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Institution 2
      </p>
    </div>
    <div
      class="cv-experience--places MuiBox-root css-0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        01/01/2002
        -
        01/01/2005
      </p>
    </div>
  </div>
</div>
`;
